crates/grafbase-sdk/wit/since_0_17_0/hooks.wit

```diff
@@ -0,0 +1,24 @@
+interface hooks {
+    use http-client.{http-method};
+    use error.{error-response};
+    use headers.{headers};
+
+    /// Hook function called when processing an incoming request
+    ///
+    /// This allows middleware to inspect and potentially modify the request
+    /// before it continues through the processing pipeline.
+    on-request: func(
+        url: string,
+        method: http-method,
+        headers: headers,
+    ) -> result<_, error-response>;
+
+    /// Hook function called when processing an outgoing response
+    ///
+    /// This allows middleware to inspect and potentially modify the response
+    /// before it is sent back to the client.
+    on-response: func(
```

Comment:
I don't think we should allow returning an `error-response` here, makes little sense to me. Execution already happened. Preventing access based on gateway response headers would be a very weird pattern at best here. If the response hook fails we should just log an error I think and that's it. 
At least I would not include this possibilty in the first hook version, I don't see any good use case for this. Do you?